app/ui/browser-modern/views/browser/tabbar/tab-contents.jsx

```diff
@@ -127,8 +128,14 @@ TabContents.propTypes = {
 function mapStateToProps(state, ownProps) {
   const page = PagesSelectors.getPageById(state, ownProps.pageId);
   const pageIsPinned = PagesSelectors.getPagePinned(state, ownProps.pageId);
+  let title = page.title || page.meta.title || page.location || 'Loading...';
```

Comment:
We don't have any prototype methods in the immutable models as of yet, but could add it for this.